pkg/client/create_manifest.go

```diff
@@ -0,0 +1,79 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/buildpacks/imgutil"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+
+	"github.com/buildpacks/pack/internal/style"
+)
+
+type CreateManifestOptions struct {
+	// Image index we want to create
+	IndexRepoName string
+
+	// Name of images we wish to add into the image index
+	RepoNames []string
+
+	// Media type of the index
+	Format types.MediaType
+
+	// true if we want to publish to an insecure registry
+	Insecure bool
+
+	// true if we want to push the index to a registry after creating
+	Publish bool
+}
+
+// CreateManifest implements commands.PackClient.
+func (c *Client) CreateManifest(ctx context.Context, opts CreateManifestOptions) (err error) {
```

Comment:
Similar comment to https://github.com/buildpacks/pack/pull/1705#discussion_r1586320093